File: LeetCode75-sols/Two_Pointers/CPP/Move_Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int i = 0;
        for (int j = 0; j < size(nums); j++) {
            if (nums[j] != 0) {
                nums[i] = nums[j];
                i++;
            }
        }
        for (int k = i; k < size(nums); k++) {
            nums[k] = 0;
        }
    }
};

File: LeetCode75-sols/Two_Pointers/Python/Move_Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        i = 0
        for j in range(len(nums)):
            if ((nums[j] != 0) & (nums[i] == 0)):
                # i is left pointer, j is right
                # Swap poisitions if true
                nums[i], nums[j] = nums[j], nums[i]
            if (nums[i] != 0):
                i += 1
                